sqlmesh/core/model/definition.py

```diff
@@ -1876,6 +1877,21 @@ def _split_sql_model_statements(
     return query, expressions[:pos], expressions[pos + 1 :]
 
 
+def _resolve_custom_session_params(
```

Comment:
If the purpose of the function is to only resolve session properties it feels somewhat counterintuitive to require 2 dicts in arguments and mutation one of them. How about the following signature:
```
def _resolve_custom_session_params(
    default: t.Optional[t.Dict[str, t.Any]], provided: t.Optional[exp.Expression]
) -> t.Optional[exp.Expression]:
    if default is not None and provided is not None:
        ...

```